library/gpg/src/modules/GPG.rb

```diff
@@ -240,22 +241,12 @@ def CreateKey
           "/gpg_tmp_exit_file"
         )
         if FileUtils.Exists(exit_file)
-          SCR.Execute(path(".target.bash"), Ops.add("rm -f ", exit_file))
+          SCR.Execute(path(".target.bash"), "/usr/bin/rm -f #{exit_file.shellescape}")
         end
 
-        command = Ops.add(
-          Ops.add(
-            Ops.add(
-              Ops.add(
-                Ops.add(Ops.add("LC_ALL=en_US.UTF-8 ", xterm), " -e \""),
-                command
-              ),
-              "; echo $? > "
-            ),
-            exit_file
-          ),
-          "\""
-        )
+        command = "LC_ALL=en_US.UTF-8 #{xterm.shellescape} -e " \
```

Comment:
That appears overkill to me. That `xterm` variable is really a constant that is set at the start of this function, then it checks if that thing is available (which could also be refactored to make this much better readable and maintainable). No different value is ever used or allowed here. So I'd vote for either simply using a string literal `/usr/bin/xterm` in both of the two (no more!) cases where it's used or at least use a constant `XTERM` instead. This can clearly never have a value that needs to be shell-escaped.